core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -433,7 +434,7 @@ public void run() {
               observer.onHeaders(headers);
             } catch (Throwable t) {
               cancel();
-              throw Throwables.propagate(t);
+              Throwables.propagate(t);
```

Comment:
Obvious, but possibly confusing, since the "throw" will never actually be executed.  I think just having the word "Throwables" in the line makes it pretty clear whats about to happen.  Also, considering how widely Guava is used, I don't think it's overly surreptitious. 

If you feel strongly about it I will revert it.